Sources/Sentry/SentryCrashIntegration.m

```diff
@@ -173,19 +173,25 @@ - (void)configureScope
             [outerScope setContextValue:appData forKey:@"app"];
 
             [outerScope addScopeListener:^(SentryScope *_Nonnull scope) {
-                NSMutableDictionary<NSString *, id> *userInfo =
-                    [[NSMutableDictionary alloc] initWithDictionary:[scope serialize]];
-
-                // SentryCrashReportConverter.convertReportToEvent needs the release name and the
-                // dist of the SentryOptions in the UserInfo. When SentryCrash records a crash it
-                // writes the UserInfo into SentryCrashField_User of the report.
-                // SentryCrashReportConverter.initWithReport loads the contents of
-                // SentryCrashField_User into self.userContext and convertReportToEvent can map the
-                // release name and dist to the SentryEvent. Fixes GH-581
-                userInfo[@"release"] = self.options.releaseName;
-                userInfo[@"dist"] = self.options.dist;
-
-                [SentryCrash.sharedInstance setUserInfo:userInfo];
+                // The serialization of the scope and synching it to SentryCrash can use quite some
+                // CPU time. We want to make sure that this doesn't happen on the main thread. We
+                // accept the tradeoff that in case of a crash the scope might not be 100% up to
+                // date over blocking the main thread.
+                [self.dispatchQueueWrapper dispatchAsyncWithBlock:^{
```

Comment:
Did you consider having a flag to control this? I wonder if this could be a opt-in or opt-out.
I guess we'll see if we start receiving feedback of _missing scope data on crashes_ but this is sort of a breaking change.

Reply:
This method is called for every change on the scope. So the crash has to happen immediately after you change the scope. I think not blocking the main thread has a way higher priority than having the scope 100% up to date all the time.  I don't think we should add a flag for this. What do you think @marandaneto and @HazAT?

Reply:
definitely doing it on the UI is not good.
losing scope's data is also not good for hard crashes.

would be possible to introduce a sort of synchronization mechanism that if a hard crash happens, we'd `await` this job to be done before crashing the app?